rskj-core/src/main/java/co/rsk/cli/tools/ExecuteBlocks.java

```diff
@@ -46,7 +49,12 @@ public static void execute(String[] args, BlockExecutor blockExecutor, BlockStor
             Block block = blockStore.getChainBlockByNumber(n);
             Block parent = blockStore.getBlockByHash(block.getParentHash().getBytes());
 
-            blockExecutor.execute(block, parent.getHeader(), false, false);
+            BlockResult blockResult = blockExecutor.execute(block, parent.getHeader(), false, false);
+
+            if (!Arrays.equals(blockResult.getFinalState().getHash().getBytes(), block.getStateRoot())) {
```

Comment:
This check will fail for blocks before 1591000 because those blockstate roots are either not validated, or converted with TrieConeverter